### ISSUE TITLE
Tailwind v3.1.5 CLI fails to install static style sheets on ubuntu stable images

### ISSUE THREAD

**Issue by nag763** [tailwindlabs/tailwindcss#8819]
Hello everyone,

I just saw this bug while trying to deploy my application in PROD. My stylesheet is generated exclusively with the tailwind CLI (installed via npx), and since I didn't specify any version on my scripts, I could notice the following regression :

**What version of Tailwind CSS are you using?**

Reproduces the bugs : 3.1.5
Last known working : 3.1.4

**What build tool (or framework if it abstracts the build tool) are you using?**

I was using the Rust [1.62](https://hub.docker.com/layers/rust/library/rust/1.62.0/images/sha256-5709afe04a23c0a447b02077b3ab8ff3d9458e80e1f9898e40873df36a34981b?context=explore) docker image to build my style sheet, but after moving to the [22.04](https://hub.docker.com/layers/ubuntu/library/ubuntu/22.04/images/sha256-cd3d86f1fb368c6a53659d467560010ab9e0695528127ea336fe32f68f7ba09f?context=explore) ubuntu's docker image, I could see it was reproducible as well.

**What version of Node.js are you using?**

On ubuntu 22.04 : 
* npm -v :arrow_right:  v8.5.1
* node -v :arrow_right:  v12.22.9

I am not sure if it could help, but the issue isn't reproducible on my main environment with the following versions :
* npm -v :arrow_right: 6.14.17
* node -v :arrow_right: v14.20.0

**What browser are you using?**

Not related to the browser, error is thrown on install

**What operating system are you using?**

A cat on /etc/os-release on the Ubuntu latest Docker image shows : 

```
PRETTY_NAME="Ubuntu 22.04 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

**Reproduction URL**

The following Dockerfile is enough 

```docker
FROM ubuntu:latest

RUN apt update; apt install npm -y

WORKDIR /usr/src/myworkdir

COPY ./ .

RUN cat /etc/os-release

RUN npm -v

RUN node -v

# The commands below have been simplified in order to ignore unecessary arguments/ input parameters

# Pass with success
RUN npx tailwindcss@3.1.4 -o tailwind_3_1_4.css

# Fails
RUN npx tailwindcss@3.1.5 -o tailwind_3_1_5.css
```

**Describe your issue**

The problem is pretty simple, when we try to generate the stylesheet output in a file, the following error is thrown on version 3.1.5 : 
```
npm WARN exec The following package was not found and will be installed: tailwindcss@3.1.5
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'postcss-load-config@4.0.1',
npm WARN EBADENGINE   required: { node: '>= 14' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'yaml@2.1.1',
npm WARN EBADENGINE   required: { node: '>= 14' },
npm WARN EBADENGINE   current: { node: 'v12.22.9', npm: '8.5.1' }
npm WARN EBADENGINE }
/root/.npm/_npx/b1deed7f46137711/node_modules/yaml/dist/compose/composer.js:33
                if (prelude[i + 1]?.[0] !== '#')
                                   ^

SyntaxError: Unexpected token '.'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/root/.npm/_npx/b1deed7f46137711/node_modules/yaml/dist/index.js:3:16)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10
```

From my understanding, it is indeed linked with the version of node being too low, but I find it disturbing that it is not mentionned in the repository 3.1.5 release draft ([here](https://github.com/tailwindlabs/tailwindcss/releases/tag/v3.1.5)) besides of being a minor version change.
I believe I am not the only one to use the tool this way, hence why I prefered reporting this problem.

Thanks in advance for the hardwork you are doing and you have been doing on this nice tool that is Tailwind :smile:

**Comment by timkley** [comment]
@adamwathan this seems to be introduced by [this pull request](https://github.com/tailwindlabs/tailwindcss/pull/8686). `postcss-load-config` dropped support for Node 10 and 12 with this major release. I think this automated pull request was merged by mistake and should be reverted to 3.1.4.

**Comment by thecrypticace** [comment]
I've pushed a fix for this and our insiders build should be fixed (`npm install tailwindcss@insiders`) but am going to leave this open until we tag a release today. Sorry for the trouble — we didn't catch that v4 dropped Node v12 support.

**Comment by thecrypticace** [comment]
Just published v3.1.6 which should fix the issues for you! Feel free to update this issue if you run into any more problems.

**Comment by darnocian** [comment]
This problem seems to exist again in v3.3.2.  Looked fine in v3.3.1.
